ansible/files/admin_api_scripts/activate_read_replica.sh

```diff
@@ -0,0 +1,41 @@
+#! /usr/bin/env bash
+
+function read_replica_preparations {    
+    # Clear everything beforehand
+    if [[ -d /tmp/wal_fetch_dir ]]; then
+        rm -rf /tmp/wal_fetch_dir
+    fi
+    
+    mkdir /tmp/wal_fetch_dir
+    chown postgres:postgres /tmp/wal_fetch_dir
+    chmod 770 /tmp/wal_fetch_dir
+    
+    backup_name=$1
+    
+    # Stop database and empty it
+    systemctl stop postgresql
+    rm -rf /var/lib/postgresql/data/*
+
+    # Download base backup
+    wal-g backup-fetch /var/lib/postgresql/data "$backup_name" --config /etc/wal-g/config.json
```

Comment:
is this blocking? will it cause the req to adminapi to timeout?

Reply:
It won't. The function will be invoked [here](https://github.com/supabase/postgres/blob/575e85eb28adeb2e45b62ba61f0403f086f3bab4/ansible/files/admin_api_scripts/activate_read_replica.sh#L40) and run on the background.